pkg/core/state/policy_executions.go

```diff
@@ -0,0 +1,93 @@
+type CheckResult struct {
+	ExecutionId        uuid.UUID `db:"execution_id"`
+	ExecutionTimestamp time.Time `db:"execution_timestamp"`
+	Name               string    `db:"name"`
+	Selector           string    `db:"selector"`
+	Description        string    `db:"description"`
+	Status             string    `db:"status"`
+	Result             string    `db:"result"`
+	Error              string    `db:"error"`
+}
+
+// PolicyExecution represents a single policy execution and their summary
+type PolicyExecution struct {
+	Id           uuid.UUID `db:"id"`
+	Timestamp    time.Time `db:"timestamp"`
+	Scheme       string    `db:"scheme"`
+	Location     string    `db:"location"`
+	PolicyName   string    `db:"policy_name"`
+	Selector     string    `db:"selector"`
+	Sha256Hash   string    `db:"sha256_hash"`
+	Version      string    `db:"version"`
+	ChecksTotal  int       `db:"checks_total"`
+	ChecksFailed int       `db:"checks_failed"`
+	ChecksPassed int       `db:"checks_passed"`
+}
+
+// CreateCheckResult inserts a check result in the database
+func (c *Client) CreateCheckResult(ctx context.Context, cr *CheckResult) error {
+	if !c.StoreRunResults {
+		return nil
+	}
+	q := goqu.Dialect("postgres").Insert("cloudquery.check_results").Rows(cr)
+	sql, args, err := q.ToSQL()
+	if err != nil {
+		return err
+	}
+	return c.db.Exec(ctx, sql, args...)
+}
+
+// CreatePolicyExecution inserts a policy execution in the database
+func (c *Client) CreatePolicyExecution(ctx context.Context, pe *PolicyExecution) (*PolicyExecution, error) {
+	if !c.StoreRunResults {
+		return pe, nil
+	}
+	var err error
+	if pe.Id, err = uuid.NewUUID(); err != nil {
+		return nil, err
+	}
+	pe.Timestamp = time.Now()
+	q := goqu.Dialect("postgres").Insert("cloudquery.policy_executions").Rows(pe)
+	sql, args, err := q.ToSQL()
+	if err != nil {
+		return nil, err
+	}
+	return pe, c.db.Exec(ctx, sql, args...)
+}
+
+// PrunePolicyExecutions deletes old policy executions in the database
+func (c *Client) PrunePolicyExecutions(ctx context.Context, pruneBefore time.Time) error {
+	q := goqu.Dialect("postgres").Delete("cloudquery.policy_executions").Where(goqu.Ex{
+		"timestamp": goqu.Op{"lt": pruneBefore},
+	})
+	sql, args, err := q.ToSQL()
+	if err != nil {
+		return err
+	}
+	return c.db.Exec(ctx, sql, args...)
+}
+
+// UpdatePolicyExecution updates a policy execution in the database
+func (c *Client) UpdatePolicyExecution(ctx context.Context, pe *PolicyExecution) error {
+	if !c.StoreRunResults {
+		return nil
+	}
+	return c.db.Exec(ctx, `
+		UPDATE cloudquery.policy_executions
+		SET checks_total = (SELECT COUNT(execution_id) FROM cloudquery.check_results WHERE execution_id = $1),
+		checks_failed = (SELECT COUNT(execution_id) FROM cloudquery.check_results WHERE execution_id = $2 AND status = 'failed'),
+		checks_passed = (SELECT COUNT(execution_id) FROM cloudquery.check_results WHERE execution_id = $3 AND status = 'passed')
+		WHERE id = $4
+	`, pe.Id, pe.Id, pe.Id, pe.Id)
```

Comment:
Always use `$1` (in multiple places in the query) so you can do a single `pe.Id` param?